### PR TITLE
Add fortran test

### PR DESCRIPTION
**Pull request by mauneyc-LANL**
## PR Summary

Adds a test to compile and run a Fortran program that uses the `singularity-eos` module.

This test simply compiles and executes, and does not do any validation. That is, it just confirms that the Fortran interface works.

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file

### CODE REVIEW

**Comment by mauneyc-LANL** [comment]
@jonahm-LANL @Yurlungur 
I'm hitting some confustion. Can I build `-DSINGULARITY_BUILD_CLOSURE=OFF -DSINGULARITY_USE_FORTRAN=ON` ? Or, is it the case that `SINGULARITY_BUILD_CLOSE=ON` iff `SINGULARITY_USE_FORTRAN=ON`?

**Comment by dholladay00** [comment]
Initially, the fortran interface was only needed when closure was also needed and it relied on the existence of certain functions. With the SAP integration, this needs to now be disentangled with the use of the preprocessor I guess. One should be able to (perhaps with your changes) enable fortran without enabling closure or so that is the new intent given the SAP integration.

File: CMakeLists.txt [hunk 1]
```diff
@@ -422,11 +422,19 @@ set(with_warnings "$<NOT:$<BOOL:${SINGULARITY_HIDE_MORE_WARNINGS}>>")
 set(with_better_flags "$<BOOL:${SINGULARITY_BETTER_DEBUG_FLAGS}>")
 
 set(xlfix "$<${with_xlcxx}:-std=c++1y;-qxflag=disable__cplusplusOverride>")
+# default generators do not invoke preprocessor on fortran sources,
+# these add the appropriate flag to do this
+set(pp_fort "$<$<COMPILE_LANG_AND_ID:Fortran,GNU>:-cpp>")
+set(pp_fort "${pp_fort}$<$<COMPILE_LANG_AND_ID:Fortran,Intel>:-fpp>")
+set(pp_fort "${pp_fort}$<$<COMPILE_LANG_AND_ID:Fortran,PGI>:-Mpreprocess>")
+set(pp_fort "${pp_fort}$<$<COMPILE_LANG_AND_ID:Fortran,Cray>:-eZ>")
+set(pp_fort "${pp_fort}$<$<COMPILE_LANG_AND_ID:Fortran,XL>:-qpreprocess>")
```
Comment (Yurlungur) [inline CMakeLists.txt:431]: because these are generators, do they do anything if fortran is disabled?

**Review by Yurlungur — APPROVED** [review]
I like this. good to have a test in place.

**Comment by Yurlungur** [comment]
> Initially, the fortran interface was only needed when closure was also needed and it relied on the existence of certain functions. With the SAP integration, this needs to now be disentangled with the use of the preprocessor I guess. One should be able to (perhaps with your changes) enable fortran without enabling closure or so that is the new intent given the SAP integration.

How tightly coupled are these still, @dholladay00 ? Can one compile `singularity_eos.cpp` without mixed cell closures?

**Comment by Yurlungur** [comment]
@mauneyc-LANL the minimal test is failing because eigen is being included somewhere, despite closures being disabled. Might be a missing #ifdef in singularity_eos.cpp.

File: CMakeLists.txt (hunk 1 above)
Comment (mauneyc-LANL) [reply CMakeLists.txt:431]: They only apply to fortran source

**Comment by mauneyc-LANL** [comment]
> @mauneyc-LANL the minimal test is failing because eigen is being included somewhere, despite closures being disabled. Might be a missing #ifdef in singularity_eos.cpp.

There is a conflict b/w `BUILD_CLOSURE=OFF` and 'USE_FORTRAN=ON`.  `BUILD_CLOSURE=OFF` disables including `eos/singularity-eos.cpp`, however there are fortran interface functions that need to link to functions in that file.

**Comment by Yurlungur** [comment]
@mauneyc-LANL what's the status of this?

**Comment by dholladay00** [comment]
I recently added a very similar fortran test. Closing this for now.

**Pull request closed without merging**